### PR TITLE
Fixing bug for K>1 and more than two classes / Adding basic utils test functionality

### PR DESCRIPTION
**Pull request by rengglic**
You may want to test this on a GPU as well and maybe add a tensorflow_hub:expect_sklearn_installed rule or sklearn as a dependency in the BUILD file (needed to verify the correctness of the KNN accuracy calculation). Running matmul using TF on CPU only is really slow compared to numpy. Therefore a check was introduced in the utils.py to choose the faster library in any case...

### CODE REVIEW

**Comment by rengglic** [comment]
@googlebot I signed it!

**Comment by andresusanopinto** [comment]
Thanks @rengglic. This has been merged as https://github.com/tensorflow/hub/commit/20f388cfe01635a2073723c5c4ffd6dd7a87fbb2
